core/domain/topic_services.py

```diff
@@ -0,0 +1,175 @@
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.]
+
+"""Commands for operations on topics, and related models."""
+
+import logging
+
+from core.domain import role_services
+from core.domain import topic_domain
+from core.domain import user_services
+from core.platform import models
+
+(topic_models,) = models.Registry.import_models([models.NAMES.topic])
+
+
+def get_topic_rights_from_model(topic_rights_model):
+    """Constructs a TopicRights object from the given topic rights model.
+
+    Args:
+        topic_rights_model: TopicRightsModel. Topic rights from the
+            datastore.
+
+    Returns:
+        TopicRights. The rights object created from the model.
+    """
+
+    return topic_domain.TopicRights(
+        topic_rights_model.id,
+        topic_rights_model.manager_ids
+    )
+
+
+def save_topic_rights(topic_rights, committer_id, commit_message, commit_cmds):
+    """Saves a TopicRights domain object to the datastore.
+
+    Args:
+        topic_rights: TopicRights. The rights object for the given
+            topic.
+        committer_id: str. ID of the committer.
+        commit_message: str. Descriptive message for the commit.
+        commit_cmds: list(dict). A list of commands describing what kind of
+            commit was done.
+    """
+
+    model = topic_models.TopicRightsModel.get(topic_rights.id, strict=False)
+
+    model.manager_ids = topic_rights.manager_ids
+
+    model.commit(committer_id, commit_message, commit_cmds)
+
+
+def create_new_topic_rights(topic_id, committer_id):
+    """Creates a new topic rights object and saves it to the datastore.
+
+    Args:
+        topic_id: str. ID of the topic.
+        committer_id: str. ID of the committer.
+    """
+
+    topic_rights = topic_domain.TopicRights(topic_id, [])
+    commit_cmds = [{'cmd': topic_domain.CMD_CREATE_NEW}]
+
+    topic_models.TopicRightsModel(
+        id=topic_rights.id,
+        manager_ids=topic_rights.manager_ids
+    ).commit(committer_id, 'Created new topic rights', commit_cmds)
+
+
+def get_topic_rights(topic_id, strict=True):
+    """Retrieves the rights object for the given topic.
+
+    Args:
+        topic_id: str. ID of the topic.
+
+    Returns:
+        TopicRights. The rights object associated with the given topic.
+
+    Raises:
+        EntityNotFoundError. The topic with ID topic_id was not
+            found in the datastore.
+    """
+
+    model = topic_models.TopicRightsModel.get(topic_id, strict=strict)
+
+    if model is None:
+        return None
+
+    return get_topic_rights_from_model(model)
+
+
+def check_can_manage_topic(user, topic_rights):
```

Comment:
Why "manage" rather than "edit"? (What's the difference?)

Reply:
Changed this. I think manage would contain a lot more actions than just edit. This function actually only checks the latter. Renamed it.